staging/src/k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap/default.go

```diff
@@ -314,19 +315,38 @@ var (
 			ResourceRules: []flowcontrol.ResourcePolicyRule{
 				resourceRule(
 					[]string{"get", "create", "update"},
-					[]string{corev1.GroupName},
-					[]string{"endpoints", "configmaps"},
-					[]string{"kube-system"},
+					[]string{coordinationv1.GroupName},
+					[]string{"leases"},
+					[]string{flowcontrol.NamespaceEvery},
 					false),
+			},
+		},
+	)
+	// We add an explicit rule for endpoints-controller with high precedence
+	// to ensure that those calls won't get caught by the following
+	// <workload-leader-election> flow-schema.
+	//
+	// TODO(#80289): Get rid of this rule once we get rid of support for
+	//   using endpoints and configmaps objects for leader election.
+	SuggestedFlowSchemaEndpointsController = newFlowSchema(
+		"endpoint-controller", "workload-high", 150,
+		flowcontrol.FlowDistinguisherMethodByUserType,
+		flowcontrol.PolicyRulesWithSubjects{
+			Subjects: append(
+				users(user.KubeControllerManager),
+				kubeSystemServiceAccount("endpoint-controller", "endpointslicemirroring-controller")...),
+			ResourceRules: []flowcontrol.ResourcePolicyRule{
 				resourceRule(
 					[]string{"get", "create", "update"},
-					[]string{coordinationv1.GroupName},
-					[]string{"leases"},
+					[]string{corev1.GroupName},
+					[]string{"endpoints"},
 					[]string{flowcontrol.NamespaceEvery},
 					false),
 			},
 		},
 	)
+	// TODO(#80289): Get rid of this rule once we get rid of support for
+	//   using endpoints and configmaps objects for leader election.
 	SuggestedFlowSchemaWorkloadLeaderElection = newFlowSchema(
 		"workload-leader-election", "leader-election", 200,
```

Comment:
do we still need this rule, how are we distinguishing it from a regular configmap crud?

Reply:
i see it now, `endpoint-controller` flowschema isolates endpoint crud. does this mean no other entity is going to do crud on endpoint like a network component or a user via kubectl?

Reply:
No, the point of the `endpoint-controller` flow schema is to properly classify that traffic rather than let the `workload-leader-election` flow schema mis-classify it.

Which prompts me to ask, why is `endpoint-controller` so narrow?  Why not catch all the controller-manager CRUD of endpoints and configmaps?